### PR TITLE
Licensing: Document new v7.4 options and APIs

### PR DESCRIPTION
**Pull request by sakjur**
**What this PR does / why we need it**:

- Add docs for new `license_text` option.
- Add docs for new `auto_refresh_license` option.
- Add docs for removing license from database API.
- Add docs for manual refresh of license option.
- Minor polishing of the docs surrounding my changes.

**Note:** This PR does not represent the full extent of changes done to licensing in v7.4.0, see #30216.
**Note 2:** ~This is a draft for now, as `auto_refresh_license` is still called `auto_renew_license`.~

### CODE REVIEW

File: docs/sources/enterprise/activate-license.md [hunk 1]
```diff
@@ -37,6 +45,19 @@ There are two different ways to add the license file to a Grafana instance:
...
+
+  ```bash
+  [enterprise]
+  license_text = eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0aGlzIjoiaXMiLCJub3QiOiJhIiwidmFsaWQiOiJsaWNlbnNlIn0.bxDzxIoJlYMwiEYKYT_l2s42z0Y30tY-6KKoyz9RuLE
+  ```
+  
+  This option can be set using the `GF_ENTERPRISE_LICENSE_TEXT`
+  environment variable.
+
 ## Step 3. Ensure that the license file's root url matches the root_url configuration option
```
Comment (oddlittlebird) [inline docs/sources/enterprise/activate-license.md:61]:
```suggestion
## Step 3. Ensure that the license file's root URL matches the root_url configuration option
```

File: docs/sources/enterprise/activate-license.md [hunk 2]
```diff
@@ -55,3 +76,8 @@ Or with an environment variable:
 ```
 GF_SERVER_ROOT_URL=https://grafana.blah.com/
 ```
+
+## Step 4. Restart Grafana
+
+To finalize the installation of Grafana Enterprise, restart Grafana to
```
Comment (oddlittlebird) [inline docs/sources/enterprise/activate-license.md:82]: I suggest adding the command(s) to do that in this step.

File: docs/sources/enterprise/enterprise-configuration.md [hunk 3]
```diff
@@ -9,6 +9,31 @@ weight = 300
...
+## [enterprise]
+
+### license_path
+
+Local filesystem path to Grafana Enterprise's license file.
+Defaults to `<paths.data>/license.jwt`.
+
+### license_text
+
+> Only available in Grafana Enterprise v7.4+.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/enterprise-configuration.md:21]:
```suggestion
> **Note:** Available in Grafana Enterprise v7.4+.
```

File: docs/sources/enterprise/enterprise-configuration.md [hunk 4]
```diff
@@ -9,6 +9,31 @@ weight = 300
...
+
+When set to the text representation (i.e. content of the license file)
+of the license, Grafana will evaluate and apply the given license to
+the instance.
+
+### auto_refresh_license
+
+> Only available in Grafana Enterprise v7.4+.
```
Comment (oddlittlebird) [inline docs/sources/enterprise/enterprise-configuration.md:29]:
```suggestion
> **Note:** Available in Grafana Enterprise v7.4+.
```

**Comment by oddlittlebird** [comment]
@achatterjee-grafana and @osg-grafana, leaving this for you two aside from some obvious suggestions that I made.

File: docs/sources/http_api/licensing.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+description = "Grafana Licensing HTTP API"
+keywords = ["grafana", "http", "documentation", "api", "licensing", "enterprise"]
+aliases = ["/docs/grafana/latest/http_api/licensing/"]
++++
+
+> Licensing is only available in Grafana Enterprise. Read more about [Grafana Enterprise]({{< relref "../enterprise" >}}).
+
+## Manually force license refresh
+
+> Only available in Grafana Enterprise v7.4+.
```
Comment (sakjur) [inline docs/sources/http_api/licensing.md:12]:
```suggestion
> **Note:** Only available in Grafana Enterprise v7.4+.
```

File: docs/sources/http_api/licensing.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+Status Codes:
+
+- **200** - OK
+- **401** - Unauthorized
+- **403** - Access denied
+
+## Remove license from database
+
+> Only available in Grafana Enterprise v7.4+.
```
Comment (sakjur) [inline docs/sources/http_api/licensing.md:62]:
```suggestion
> **Note:** Only available in Grafana Enterprise v7.4+.
```

File: docs/sources/enterprise/activate-license.md (hunk 2 above)
Comment (sakjur) [reply docs/sources/enterprise/activate-license.md:82]: That's makes sense. What systems should we at least target when writing that instruction?

I can add `sudo systemctl restart grafana-server` as the baseline command for people who have followed the most basic installing Grafana as a service on one of the Linux distributions we provide packages for, but when we start coming into Docker, Kubernetes, Windows, macOS, and whatnot territory that'd be a topic that should be covered in its own right in a separate PR by someone who's less of a wannabe system administrator than I am.

Comment (oddlittlebird) [reply docs/sources/enterprise/activate-license.md:82]: I would duplicate what we have in the installation instructions. 

Do you really think there are enough different ways to restart Grafana that they deserve a topic? If so, then I will add it to our backlog.

File: docs/sources/enterprise/activate-license.md [hunk 7]
```diff
@@ -11,22 +11,30 @@ Follow these steps to activate your Grafana Enterprise license:
...
+  This is the preferred option for single instance installations of
+  Grafana Enterprise. 
 
   1. Sign in as a Grafana server admin.
   1. Navigate to **Server Admin > Upgrade** within Grafana. 
   1. Click **Upload license token file**.
   1. Select your license file, and upload it.
 
-- Option 2: Place the *license.jwt* file in Grafana's data folder.
+### Place the *license.jwt* file in Grafana's data folder
```
Comment (oddlittlebird) [inline docs/sources/enterprise/activate-license.md:35]:
```suggestion
### Place the license.jwt file in Grafana's data folder
```

File: docs/sources/enterprise/activate-license.md [hunk 8]
```diff
@@ -37,7 +45,20 @@ There are two different ways to add the license file to a Grafana instance:
 
   This setting can also be set with an environment variable, which is useful if you're running Grafana with Docker and have a custom volume where you have placed the license file. In this case, set the environment variable `GF_ENTERPRISE_LICENSE_PATH` to point to the location of your license file.
 
-## Step 3. Ensure that the license file's root url matches the root_url configuration option
+### Set the content of the license file on the *enterprise.license_text* configuration option
```
Comment (oddlittlebird) [inline docs/sources/enterprise/activate-license.md:48]: Avoid italics in headings. You can just remove the file name from the heading. It is specified in the prose.

File: docs/sources/http_api/licensing.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+description = "Grafana Licensing HTTP API"
+keywords = ["grafana", "http", "documentation", "api", "licensing", "enterprise"]
+aliases = ["/docs/grafana/latest/http_api/licensing/"]
++++
+
+> Licensing is only available in Grafana Enterprise. Read more about [Grafana Enterprise]({{< relref "../enterprise" >}}).
+
+## Manually force license refresh
+
+> **Note:** Only available in Grafana Enterprise v7.4+.
```
Comment (oddlittlebird) [inline docs/sources/http_api/licensing.md:12]: We are trying to avoid "only" in these notes, because it can cause confusion. Please remove that word in all cases and just say, "Available in..."

File: docs/sources/enterprise/activate-license.md (hunk 2 above)
Comment (sakjur) [reply docs/sources/enterprise/activate-license.md:82]: I couldn't find instructions for restarting in the installation instructions or anywhere in our docs, but I'll squeeze in a `systemctl` reference so that the user has _some_ level of guidance here. Thanks for pointing it out, I remember when being new and getting a "do X" instruction and being so frustrated that barely anyone would bother sharing even the slightest hint to where to look for instructions on how to do X, and here I'm just as guilty of making the same assumptions about people's skill levels 😅

As for if restart instructions deserve a topic on their own, maybe. We support at least Windows, macOS, Linux, and Docker. The process to install, set up as service, configure, upgrade, restart, and uninstall is different for each of those, and it'd be convenient to be able to refer to a central documentation where the user can see OS specific instructions.

If we decide that we can document for say Linux and then ask readers to translate that for them, I'd be ok with that too. To me, what to target when documenting is a design decision that I'd defer to you.

I don't know enough to provide details for Windows without help, though — and I think it'd be useful to document these key idiosyncrasies by having more comprehensive Windows, Mac, Linux (we list a lot of Linux specifics already, but not in a "Linux overview" page style), and Docker guides covering system operations beyond just installations.

tl;dr: Yes. I do. What I'm not sure about is what priority it should have.

File: docs/sources/enterprise/activate-license.md (hunk 8 above)
Comment (sakjur) [reply docs/sources/enterprise/activate-license.md:48]:
```suggestion
### Set the content of the license file as a configuration option
```

**Review by osg-grafana — APPROVED** [review]
Approving to get this in. Can make more tweaks later as needed.

File: docs/sources/http_api/licensing.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (sakjur) [reply docs/sources/http_api/licensing.md]: These were changed in 0c0cd28
